pages/dishes/index.js

```diff
@@ -1,3 +1,105 @@
+import {
+  Wrapper,
+  List,
+  ListItem,
+  ListAddButton,
+  AddDishForm,
+  Input,
+  BackButton,
+  Label,
+  SubmitButton,
+  CloseFormButton,
+  ButtonWrapper,
+} from "../../components/DishesPage/styles.js";
+import useCalorieStore from "../../utils/useCalorieStore.js";
+import Link from "next/link.js";
+import { useState } from "react";
+
 export default function DishesPage() {
-  return null;
+  const { dishes, addDish } = useCalorieStore();
+
+  const [formVisibility, toggleFormVisibility] = useState(false);
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formdata = new FormData(event.target);
+    const data = Object.fromEntries(formdata);
+    const calories = parseInt(data.calories);
+    const mass = parseInt(data.mass);
+    const proteins = parseInt(data.proteins);
+    const carbs = parseInt(data.carbs);
+    addDish(
+      data.meal,
+      calories,
+      mass ? mass : null,
+      proteins ? proteins : null,
+      carbs ? carbs : null,
+      data.notes ? data.notes : null
+    );
+    toggleFormVisibility(false);
+    event.target.reset();
+  }
+  return (
+    <Wrapper>
+      <Link href="/">
+        <BackButton>🔙</BackButton>
+      </Link>
+      <List invisible={formVisibility}>
+        {dishes.map((dish, index) => (
+          <ListItem
+            key={index}
+          >{`${dish.meal} --- ${dish.calories}kcal`}</ListItem>
+        ))}
+        <ListAddButton
+          onClick={() => {
+            toggleFormVisibility(!formVisibility);
+          }}
+        >
+          add Item
+        </ListAddButton>
+      </List>
+      <AddDishForm
+        visible={formVisibility}
```

Comment:
You can shorten your code here, since your `handleSubmit` is only taking the `event` as a prop.
This should work too:
```
onSubmit={handleSubmit}
```

Reply:
thx for the suggestion!